libcnb-data/src/buildpack/api.rs

```diff
@@ -21,31 +19,16 @@ impl TryFrom<&str> for BuildpackApi {
 
     fn try_from(value: &str) -> Result<Self, Self::Error> {
         // We're not using the `semver` crate, since it only supports non-range versions of form `X.Y.Z`.
-        lazy_static! {
-            static ref RE: Regex = Regex::new(r"^(?P<major>\d+)(\.(?P<minor>\d+))?$").unwrap();
-        }
-
-        if let Some(captures) = RE.captures(value).unwrap_or_default() {
-            if let Some(major) = captures.name("major") {
-                // these should never panic since we check with the regex unless it's greater than
-                // `std::u32::MAX`
-                let major = major
-                    .as_str()
-                    .parse::<u32>()
-                    .map_err(|_| Self::Error::InvalidBuildpackApi(String::from(value)))?;
-
-                // If no minor version is specified default to 0.
-                let minor = captures
-                    .name("minor")
-                    .map_or("0", |s| s.as_str())
-                    .parse::<u32>()
-                    .map_err(|_| Self::Error::InvalidBuildpackApi(String::from(value)))?;
-
-                return Ok(Self { major, minor });
-            }
-        }
-
-        Err(Self::Error::InvalidBuildpackApi(String::from(value)))
+        // If no minor version is specified, it defaults to `0`.
+        let (major, minor) = value.split_once('.').unwrap_or((value, "0"));
```

Comment:
Thank you - good to see for comparison! 

Things I like about that version:
- less repetition of the `map_err`
- the parsing failure cases are less hidden (vs try operator inside the creation of the return value in the other version)

Things I like less:
- the construction of the return value for the common case is hidden in the middle of the combinator chain
- nested closures
- takes several more seconds of reading to understand (for me at least; I imagine someone more familiar with combinators less so, but I suspect even then still slightly slower than the other version)

I'm probably slightly leaning towards the current version, but not so much as to oppose the above approach if you/someone else feels otherwise :-) (But a good learning opportunity regardless!)

Reply:
Would you like me to change `let (major, minor)` to `let (major_string, minor_string)` in this PR, similar to the above? (Although I guess if we're being pedantic, it's a string slice, so `major_str` etc?)

Reply:
> Would you like me to change let (major, minor) to let (major_string, minor_string) in this PR

It's fine the way it is. :)

> I'm probably slightly leaning towards the current version

Go ahead and merge if you feel that way. The comment above really was just a show-and-tell, not a request for changes.